core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -855,7 +855,9 @@ private[spark] class Executor(
    * Schedules a task to report heartbeat and partial metrics for active tasks to driver.
    */
   private def startDriverHeartbeater(): Unit = {
-    val intervalMs = conf.getTimeAsMs("spark.executor.heartbeatInterval", "10s")
+    val intervalMs = conf
```

Comment:
I agree with the change. In branch 2.4 this is accessed both with getTimeAsSeconds and getTimeAsMs. The only question is which interpretation was intended? it's not a best practice to not specify the units, but hey it can happen.

In master, this is interpreted as milliseconds by default. I think we should instead switch how it's read in SparkConf instead?

Reply:
okay, seems more reasonable as it also avoids  the scenario @MaxGekk mentioned. Will modify the PR accordingly